python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -0,0 +1,306 @@
+
+    @property
+    def columns_in_ref(self) -> Set[str]:
+        return {self.column_ref_flux}
+
+    @property
+    def columns_in_target(self) -> Set[str]:
+        return set(self.columns_target_flux).union(set(self.columns_target_flux_err))
+
+
+class DiffMatchedTractCatalogConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=DiffMatchedTractCatalogConnections,
+):
+    column_matched_prefix_ref = pexConfig.Field(
+        dtype=str,
+        default='refcat_',
+        doc='The prefix for matched columns copied from the reference catalog',
+    )
+    column_ref_extended = pexConfig.Field(
+        dtype=str,
+        default='is_pointsource',
+        doc='The boolean reference table column specifying if the target is extended',
+    )
+    column_ref_extended_inverted = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc='Whether column_ref_extended specifies if the object is compact, not extended',
+    )
+    column_target_extended = pexConfig.Field(
+        dtype=str,
+        default='refExtendedness',
+        doc='The target table column estimating the extendedness of the object (0 <= x <= 1)',
+    )
+
+    @property
+    def columns_in_ref(self) -> Set[str]:
+        columns_all = [self.coord_format.column_ref_coord1, self.coord_format.column_ref_coord2,
+                       self.column_ref_extended]
+        for columns_list in (
+            (
+                self.columns_ref_copy,
+            ),
+            (x.columns_in_ref for x in self.columns_flux.values()),
+        ):
+            for columns in columns_list:
+                columns_all.extend(columns)
+
+        return set(columns_all)
+
+    @property
+    def columns_in_target(self) -> Set[str]:
+        columns_all = [self.coord_format.column_target_coord1, self.coord_format.column_target_coord2,
+                       self.column_target_extended]
+        if self.coord_format.coords_ref_to_convert is not None:
+            columns_all.extend(self.coord_format.coords_ref_to_convert.values())
+        for columns_list in (
+            (
+                self.columns_target_coord_err,
+                self.columns_target_select_false,
+                self.columns_target_select_true,
+                self.columns_target_copy,
+            ),
+            (x.columns_in_target for x in self.columns_flux.values()),
+        ):
+            for columns in columns_list:
+                columns_all.extend(columns)
+        return set(columns_all)
+
+    columns_flux = pexConfig.ConfigDictField(
+        keytype=str,
+        itemtype=MatchedCatalogFluxesConfig,
+        doc="Configs for flux columns for each band",
+    )
+    columns_ref_copy = pexConfig.ListField(
+        dtype=str,
+        default=set(),
+        doc='Reference table columns to copy to copy into cat_matched',
+    )
+    columns_target_coord_err = pexConfig.ListField(
+        dtype=str,
+        listCheck=lambda x: (len(x) == 2) and (x[0] != x[1]),
```

Comment:
I thought we had policies on not using lambdas? Are we allowed to do this?

Reply:
The dev guide says "should not be used", but I don't think a `def checkLengthTwoSet(listfield):` function would improve anything.